filebeat/input/v2/compat/compat.go

```diff
@@ -133,14 +139,44 @@ func (r *runner) Start() {
 	go func() {
 		defer r.wg.Done()
 		log.Infof("Input '%s' starting", name)
+
+		inputID := r.id
+
+		var reg *monitoring.Registry
+		var unregfunc func()
+		if !r.generatedID {
+			// There is a bit of a chicken-egg issue here. In general the pipeline
+			// client will be created before the input register its metrics.
+			// Therefore, here just the registry is created. Later, when the input
+			// register its metrics, it'll add the `id` and `input` strings to the
+			// registry, which will make the registry valid to be published in the
+			// '/inputs/' monitoring endpoint. Also, some inputs use a different name
+			// than `r.input.Name()` when registering the metrics, another reason
+			// why it does not make sense to move the call to
+			// `inputmon.NewInputRegistry` here.
+			reg = r.agent.Monitoring.Namespace.GetRegistry().
+				NewRegistry(inputmon.SanitizeID(r.id))
+			unregfunc = func() {
+				r.agent.Monitoring.Namespace.GetRegistry().
+					Remove(inputmon.SanitizeID(inputID))
+			}
+		} else {
+			// when the input has no ID, no metric is published, therefore, we
+			// use a 'discard' registry.
```

Comment:
What does a 'discard' registry mean here? It it actually a noop registry or just a registry that collects metrics and they're never reported?